crates/astria-bridge-withdrawer/src/withdrawer/submitter/mod.rs

```diff
@@ -0,0 +1,269 @@
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+
+use astria_core::protocol::transaction::v1alpha1::{
+    Action,
+    TransactionParams,
+    UnsignedTransaction,
+};
+use astria_eyre::eyre::{
+    self,
+    Context,
+};
+pub(crate) use builder::Builder;
+use prost::Message as _;
+use sequencer_client::{
+    tendermint_rpc::endpoint::broadcast::tx_commit,
+    Address,
+    SequencerClientExt as _,
+    SignedTransaction,
+};
+use signer::SequencerSigner;
+use state::State;
+use tendermint::crypto::Sha256;
+use tokio::{
+    select,
+    sync::mpsc,
+    time::Instant,
+};
+use tokio_util::sync::CancellationToken;
+use tracing::{
+    debug,
+    error,
+    info,
+    info_span,
+    instrument,
+    warn,
+    Instrument as _,
+    Span,
+};
+
+use super::state;
+
+mod builder;
+mod signer;
+
+pub(super) struct Handle {
+    pub(super) batches_tx: mpsc::Sender<(Vec<Action>, u64)>,
+}
+
+pub(super) struct Submitter {
+    shutdown_token: CancellationToken,
+    state: Arc<State>,
+    batches_rx: mpsc::Receiver<(Vec<Action>, u64)>,
+    sequencer_cometbft_client: sequencer_client::HttpClient,
+    signer: SequencerSigner,
+    sequencer_chain_id: String,
+}
+
+impl Submitter {
+    pub(super) async fn run(mut self) -> eyre::Result<()> {
+        self.state.set_submitter_ready();
+
+        let reason = loop {
+            select!(
+                biased;
+
+                () = self.shutdown_token.cancelled() => {
+                    info!("received shutdown signal");
+                    break Ok("shutdown requested");
+                }
+
+                batch = self.batches_rx.recv() => {
+                    let (actions, rollup_height) = match batch {
+                        Some((actions, rollup_height)) => (actions, rollup_height),
+                        None => {
+                            info!("received None from batch channel, shutting down");
+                            break Err("channel closed");
+                        }
+                    };
+
+                    // get nonce and make unsigned transaction
+                    let nonce = get_latest_nonce(self.sequencer_cometbft_client.clone(), self.signer.address, self.state.clone()).await?;
+                    debug!(nonce, "fetched latest nonce");
+
+                    let unsigned = UnsignedTransaction {
+                        actions,
+                        params: TransactionParams {
+                            nonce,
+                            chain_id: self.sequencer_chain_id.clone(),
```

Comment:
we should fetch the chain id from the sequencer, not have it in the config. alternatively we fetch + verify against what's in the config